pkg/admin/server/handler.go

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package server
+
+import (
+	"net/http"
+
+	"mosn.io/mosn/pkg/log"
+)
+
+// Auth contains two parts: check function and failed function.
+// if check function returns false, the Auth.Check will returns false and
+// call failed function. if failed function is nil, use default instead.
+// default function will write a http forbidden without body
+type Auth struct {
+	checkAction  func(*http.Request) bool
+	failedAction func(http.ResponseWriter)
+}
+
+func NewAuth(check func(*http.Request) bool, failed func(http.ResponseWriter)) *Auth {
+	if failed == nil {
+		failed = defaultFailedFunc
+	}
+	return &Auth{
+		checkAction:  check,
```

Comment:
judge if `check` is nil

Reply:
check is required，i think tif check is nil causes panic is acceptable.

Reply:
should panic at start up，not at running.